src/elements/atoms/Button/Button.css

```diff
@@ -51,6 +51,10 @@
 
   &--link {
     text-decoration: none;
+
+    &.button--icon {
+      width: fit-content;
```

Comment:
I like the thought here, but unfortunatly fit-content is not supported in IE11 and we will have to fix that per-project :(

https://caniuse.com/#search=fit-content